### PR TITLE
Add git/github workflow for making changes when you don't own a repository for hacktoberfest

### PR DESCRIPTION
**Pull request by taylorreiter**
I'm going to have quick GIFs outlining the process for how to do this in our intro presentation, but for anyone who needs more practice/instruction, I created this longer form lesson. I'm wondering if we should create a 2023-hackathon repo in Arcadia-Science org so I could use that as a true follow along example that people could practice with at the hackathon? Or is this good enough and people can follow along with whatever repo they want to actually contribute to?

### CODE REVIEW

File: docs/workshops/overview.md [hunk 1]
```diff
@@ -6,4 +6,5 @@ See below for lesson materials for workshops that have been held at Arcadia.
 
 | Date | Tutorial | Length | Description | Recording |
 |------|----------|--------|-------------|-----------|
+| Oct 6, 2023 | [Git & GitHub for Hacktoberfest](20231006-github-hacktoberfest/lesson.md) | optional | This lesson is an optional refresher for participants in our Hacktoberfest mini-hackathon. It covers how to make contributions to repositories that you don't own via forks and pull requests. | |
```
Comment (mertcelebi) [inline docs/workshops/overview.md:9]: I know it is not related to this PR, but can you also update the schedule on the other aug page?

File: docs/workshops/20231006-github-hacktoberfest/lesson.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
+# Using Git and GitHub to Contribute to Repositories You Don't Have Write Access To
+
+Arcadia is participating in [Hacktoberfest 2023](https://hacktoberfest.com/).
+Our goal is to give back to the open source software community that we use every day in our own scientific work.
+As part of this, we are hosting a mini-hackathon to create a space for and support others in making contribution to open source software libraries.
+We've [curated a list of bioinformatics (or adjacent) software repos](https://docs.google.com/document/d/1EjyQOwPO-Zc4aKM_mTPEFo3K97Ox3lEk-DyZxlivyVU/edit?usp=sharing) that are accepting contributions.
+The purpose the lesson below is to outline how to contribute to a GitHub repository that you don't own (e.g., you don't have write access to).
```
Comment (mertcelebi) [inline docs/workshops/20231006-github-hacktoberfest/lesson.md:7]:
```suggestion
The purpose of the lesson below is to outline how to contribute to a GitHub repository that you don't own (e.g., you don't have write access to).
```

File: docs/workshops/20231006-github-hacktoberfest/lesson.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+- create an ssh key to authenticate without a password
+
+Navigate to the [Setting Up Instructions](https://training.arcadiascience.com/workshops/20220920-intro-to-git-and-github/lesson/#setting-up) in the Intro to Git and GitHub lesson and follow along with the tutorial.
+
+## Finding repositories that accept contributions & work that needs to be done
+
+### Searching for repositories that are open to external contributions
+
+Not every open source repository is in a position to accept work from new contributors.
+Contributing to a repository that isn't accepting these types of contributions at best leads to unintegrated pull requests and at worse creates additional work and head aches for maintainers.
```
Comment (mertcelebi) [inline docs/workshops/20231006-github-hacktoberfest/lesson.md:44]:
```suggestion
Contributing to a repository that isn't accepting these types of contributions at best leads to unintegrated pull requests and at worse creates additional work and headache for maintainers.
```

File: docs/workshops/20231006-github-hacktoberfest/lesson.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+The Contributor Guidelines for a repository outline how the project would like to receive contributions.
+This is a very important document to read and to continually reference as you're making your contribution to make sure you're following the preferences of the community or project you're contributing to.
+Each project is a little different, so make sure you read this document closely for each project you plan to contribute to.
+
+In the [Arcadia-Science/2023-hackathon-practice](https://github.com/Arcadia-Science/2023-hackathon-practice) repository, the contributor guide is located in the file `CONTRIBUTING.md` and it is linked to from the `README.md`.
+
+### Finding work that needs to be completed: issues, pull requests, and repo searches
+
+The contributor guidlines document often details how new contributors can find work that needs to be done.
```
Comment (mertcelebi) [inline docs/workshops/20231006-github-hacktoberfest/lesson.md:68]:
```suggestion
The contributor guidelines document often details how new contributors can find work that needs to be done.
```

**Review by mertcelebi — APPROVED** [review]

File: docs/workshops/overview.md (hunk 1 above)
Comment (taylorreiter) [reply docs/workshops/overview.md:9]: ya i'll do it either EOD today or tomorrow!

**Pull request merged**